compiler/codegen/library.cpp

```diff
@@ -228,17 +228,22 @@ static void printMakefileLibraries(fileinfo makefile, std::string name) {
     fprintf(makefile.fptr, "%s", requires.c_str());
   }
 
-  // Erase trailing newline if needed for the workaround below.
+  if (!llvmCodegen) {
+    fprintf(makefile.fptr, " %s\n", libraries.c_str());
+    return;
+  }
+  
+  // LLVM requires a bit more work to make the GNU linker happy.
   if (libraries.size() > 0 && *libraries.rbegin() == '\n') {
-    libraries.erase(libraries.end() - 1);
+    libraries.erase(libraries.end() -1);
   }
 
-  fprintf(makefile.fptr, " %s", libraries.c_str());
-
-  // GNU linker won't be able to see config symbols without this.
-  fprintf(makefile.fptr, " %s\n\n", libname.c_str());
+  // Append the Chapel library as the last linker argument.
+  fprintf(makefile.fptr, " %s %s\n\n", libraries.c_str(), libname.c_str());
 }
```

Comment:
I'd format this as an if/else rather than "if { return }; if (other condition)".  Someone might otherwise accidentally remove the return statement later without necessarily realizing the code that followed shouldn't run in the other situation

Reply:
Ok.